Justfile

```diff
@@ -0,0 +1,84 @@
+# FIXME: some of these tests don't work very well on MacOS at the moment. Should fix
+# them at earliest convenience :)
+# Also in addition to this, the `nextest-integration` test is currently disabled on macos
+with_macos_excludes := if os() == "macos" {
+    "--exclude node-runtime --exclude runtime-params-estimator --exclude near-network --exclude estimator-warehouse"
+} else {
+    ""
+}
+nightly_flags := "--features nightly,test_features"
+bash := if os() == "macos" { "/bin/bash" } else { "/usr/bin/bash" }
+
+# all the tests, as close to CI as possible
+test: test-ci test-extra
+
+# only the tests that are exactly the same as the ones in CI
+test-ci: (nextest "stable") (nextest "nightly") check-style
+
+# tests that are as close to CI as possible, but not exactly the same code
+test-extra: check-lychee
+
+# all cargo tests, TYPE is "stable" or "nightly"
+nextest TYPE: (nextest-unit TYPE) (nextest-integration TYPE)
+
+# cargo unit tests, TYPE is "stable" or "nightly"
+nextest-unit TYPE:
+    cargo nextest run \
+        --locked \
+        --workspace \
+        --exclude integration-tests \
+        --cargo-profile dev-release \
+        --profile ci \
```

Comment:
Local development does not want `--profile ci`. It largely turns on retrying the tests, which goes in opposition with the goal that people would notice them during local development and perhaps consider fixing them.

Reply:
Hmm… so similarly, this is arguably moving away from the behaviour of CI. So I’m not sure it makes sense to remove it from the `Justfile`, if the goal of `just` is to reproduce CI behaviour locally?

Reply:
`--profile ci` is *the* hack here, necessitated only because we have spuriously failing tests that were a significant hindrance on the development velocity when CI would get stuck on them. It isn’t an ideal place to be in the first place, but we cannot really afford the merge queue kicking out PRs occasionally when a test unrelated to those PRs decides to explode.

OTOH `--profile ci` is a reason why `ed25519-dalek` upgrade required so much time to work out – the spurious failure introduced by an earlier change was hidden by the retries enabled by `--profile ci`.

Despite that I still feel that having a couple retries on a per-test basis in CI *only* is the least terrible of the options available to us given the current requirements and the state of our test suite.

I believe this is one of those things that we will have to live with being different.

Reply:
Makes sense, I added it as an environment-configured toggle :)